doc-content/enterprise-only/openshift/hacep-server-maven-proc.adoc

```diff
@@ -0,0 +1,142 @@
+----
+In this line, replace `<id_group>` with a value in the `supplemental-groups` interval and `<id_user>` with a value in the `uid-range` interval, as detected i nthe previous step, similar to the following example:
++
+----
+RUN groupadd -r app -g 1000160001 && useradd -u 1000160001 -r -g app -m -d /app -s /sbin/nologin -c "App user" app && chmod 755 /app
+----
++
+. In the `springboot` directory, enter the following commands to create the image for deployment and push it into the repository configured for your OpenShift environment:
++
+----
+oc new-build --binary --strategy=docker --name openshift-kie-springboot
+oc start-build openshift-kie-springboot --from-dir=. --follow
+----
++
+. Enter the following command to detect the name of the image that was built:
++
+----
+oc get is/openshift-kie-springboot -o template --template='{{range .status.tags}}{{range .items}}{{.dockerImageReference}}{{end}}{{end}}'
+----
++
+. Open the `kubernetes/deployment.yaml` file in a text editor.
+. Replace the existing image URL with the result of the previous command.
+. Replace `<id_iser>` with the user ID that you entered in the Docker file in a previous step.
+. Remove all characters at the end of the line starting with the `@` symbol, then add `:latest` to  the line. For example: 
++
+----
+image: image-registry.openshift-image-registry.svc:5000/hacep/openshift-kie-springboot:latest
+----
++
+. Under the `containers:` line and the `env:` line, set environment variables as in the following example:
++
+----
+containers:
+  - env:
+    - name: UPDATABLEKJAR
+      value: "true"
+    - name: KJARGAV
+      value: <GroupID>:<ArtifactID>:<Version>
+    - name: MAVEN_LOCAL_REPO
+      value: /app/.m2/repository
+    - name: MAVEN_MIRROR_URL
+      value: http://<nexus_url>/repository/maven-releases/
+    - name: MAVEN_SETTINGS_XML
+      value: /app/.m2/settings.xml
+----
++
+In this example, replace the value of the `KJARGAV` variable with the group, artifavt and version (GAV) of your KJAR service and the value of the `MAVEN_MIRROR_URL` variable with the URL to the Maven repository that contains your KJAR service.
++
+You can optionally set other variables. For a list of supported environment variables, see <<hacep-server-environment-ref>>.
+. Save the file.
++
+. Enter the following command to deploy the image:
++
+----
+oc apply -f kubernetes/deployment.yaml
+----
++
+. To update the KJAR service at any later time, complete the following steps:
+.. Upload the new KJAR service to the Maven repository.
+.. Log in to the project with the `oc` command.
+.. Use the `oc get pods` command to detect the name of the pod in which the HA CEP server is running.
+.. Enter the following command:
++
+----
+oc set env pod/<podname> KJARGAV=<gav>
+----
++
+In this command, replace `<podname>` with the name of the pod and `<gav>` with the GAV of the new KJAR service.
```

Comment:
This part is wrong to update you write a code like this with the client :
`TopicsConfig envConfig = TopicsConfig.getDefaultTopicsConfig();
        Properties props = getProperties();
        try (RemoteStreamingKieSession producer = RemoteStreamingKieSession.create(props, envConfig)){
            producer.updateKJarGAV("org.kie:fake-jar:0.1-SNAPSHOT");
        }`

Reply:
I agree with Max, user has to specify the initial kjar version as property so the system know how to start and then the request to update is a two step action:
- verify to have the new kjar on Maven repo
- invoke client API to request the update
This second command will be processes as any other request by the leader first and then by all the replicas to migrate the working memory and the kiebase to the new version.
It is not an environment variable because it needs to be processed in memory without a pod restart